app/routes/languages.rb

```diff
@@ -10,6 +10,15 @@ class Languages < Core
         erb :"languages/index", locals: { active: active, inactive: inactive }
       end
 
+      get '/repository' do
+        tracks = X::Track.all
+        active, inactive = tracks.partition(&:active?)
+        inactive.sort! { |a, b| b.problems.count <=> a.problems.count }
```

Comment:
Ruby has a handy [`sort_by!`](http://ruby-doc.org/core-2.3.1/Array.html#method-i-sort_by-21) method that can make this a bit tidier:

```ruby
inactive.sort_by! { |track| track.problems.count }
```